### ISSUE TITLE
TODO list for Fast Roping

### ISSUE THREAD

**Issue by nicolasbadano** [acemod/ACE3#3475]
ref. #3463 

Ready in 3.6.0:
- [x] Add possibility to hide either of the two gantries from `ACE_friesGantry`. Requires a change in the `model.cfg`, the selections should already be there. (#3553)
- [x] Add waypoint to fatrope AI (#3544)
- [x] Add API to fatrope AI (#3544)

For 3.7.0+:
- [ ] `checkVehicleThread` should be called in the server, because the machine that created the helicopter might disconnect, leaving the thread interrupted. 
- [ ] Handle helicopter been deleted while fastroping is occurring
- [x] Zeus module to add FRIES.
- [ ] Zeus module to make AI fastrope?
- [ ] Add loadable ropes in fast rope bags ([original feature request from AGM](https://github.com/KoffeinFlummi/AGM/issues/1784))
  Possible model found by @jokoho48: \A3\Structures_F_Heli\Items\Airport\RotorCoversBag_01_F.p3d
- [ ] Add sound while fastroping ([original AGM issue](https://github.com/KoffeinFlummi/AGM/issues/1784)) (#3517)
- [x] Add possibility to have different rope lengths
- [x] Add option to manually retract FRIES (#5533)

**Comment by Khaosmatic** [comment]
Is it possible to add a new waypoint type as that could solve the 'Zeus module to make AI fastrope?' if it's possible. Just like a transport unload but instead with the ropes.

**Comment by BaerMitUmlaut** [comment]
I don't think so. I have a function for unloading AI, but it needs some cleanup. I'll PR it when it's ready.
Here's some stuff from my list:
- [ ] Add loadable ropes in [fast rope bags](http://www.finnrappel.fi/FastRopeBag27m.JPG) ([original feature request from AGM](https://github.com/KoffeinFlummi/AGM/issues/1836))  
Possible model found by @jokoho48: `\A3\Structures_F_Heli\Items\Airport\RotorCoversBag_01_F.p3d`
- [ ] Add sound while fastroping ([original AGM issue](https://github.com/KoffeinFlummi/AGM/issues/1784))
- [ ] Add possibility to have different rope lengths
- [ ] Maybe add option to manually retract FRIES

**Comment by BaerMitUmlaut** [comment]
Just got an idea whilst doing the RHS comp:
On helos with cargo ramps we can change the user action condition to lock it open.

**Comment by kripto202** [comment]
for having ai fastrope, there is already a mod/script out already doing that, maybe you guys can ask if he could help out. Now for the sounds, that's nice and all but not really needed which should be more on the backburner in my own opinion.

**Comment by BaerMitUmlaut** [comment]
Those functions will not work with ACE3's fast roping and as I said,
>I have a function for unloading AI, but it needs some cleanup.

**Comment by BaerMitUmlaut** [comment]
From Slack: Add possibility to hide either of the two gantries from `ACE_friesGantry`. Requires a change in the `model.cfg`, the selections should already be there.

**Comment by 654wak654** [comment]
Want might to add something else to the list too. The fastrope system is somewhat 'incompatible' with the `ace_zeus_autoAddObjects` setting, because the helper objects are added to the list of editable items. Whould it be possible to remove them from curatorEditableObjects?

![](http://i.imgur.com/ZDSN1Pl.png)

**Comment by esteldunedain** [comment]
> Whould it be possible to remove them from curatorEditableObjects?

Yeah, sure

**Comment by Max255PL** [comment]
A suggestion from our pilots - ability to cut ropes more easily would be great. Either through action menu or a keybind, since you can't really take your hands off the controls and cut them through ACE interaction menu mid-hover on AFM.

**Comment by grester** [comment]
Not sure if I should make a standalone report about this but I thought it'd be better to just mention it here.
Yesterday Friday 11th March, I was training with my team the FRIES system to use on missions and we came across a bug which unfortunately I have no idea how to replicate. The helicopter in which was tested was a RHS Blackhawk with FRIES module sync'd, the server was suferring some desync which I don't know if it could be a bugging factor. As we were training to deploy troops via the ropes some players would get into an awkward situation.
The players once they'd select to fast rope, they'd be stuck to the outside of the helicopter and a rope would be deployed on them, not the helicopter, despite the helicopter had their ropes deployed, otherwise the players wouldn't be able to fast rope in the first place.
It is strange the fact that even after restarting the game those players would still have the bug which would seem the problem could be on their side as they've tried on two separate helicopters and in both cases they would get stuck. Ocasionally some players would drop from the ropes amid fast rope but I believe that was due to players fast roping to close in between each other.

**Comment by BaerMitUmlaut** [comment]
First off, the RHS Blackhawk has not been configured yet with 3.5.0 for fastroping. I know that it has the interaction point, but that is due to how the RHS configs are made and not on purpose. Until 3.5.1 you should only use vanilla helicopters.

Additionally there's a bug that occurs when more than one person fast ropes at the same time which is probably what you're experiencing here. It will be fixed with 3.5.1.

**Comment by madpat3** [comment]
i think you mixed up the gantry with the gantryReverse for the wildcat.
![20160320143259_1](https://cloud.githubusercontent.com/assets/15272840/13904830/7b3ff1c0-eead-11e5-9ce3-194bc86af859.jpg)
![20160320143316_1](https://cloud.githubusercontent.com/assets/15272840/13904833/82efe254-eead-11e5-845a-53820cfa7a0f.jpg)

**Comment by Crielaard** [comment]
Using Ace3 CBA and TFAR only.
Everyone gets stuck on the rope, detached when ropes are cut. tried 5-20m altiude. Vanilla Ghosthawk.
Result is ropes have to be cut and redeployed for every 2 guys.

**Comment by BaerMitUmlaut** [comment]
@Crielaard and @madpat3, please make separate issues, this is a todo list, not a bug list.

**Comment by Crielaard** [comment]
Ow! yeah you're right. srry

**Comment by NoBullshit** [comment]
Hello, I noticed that the AIs lost their waypoints after fastroping... They wait on the ground forever. I also tried as group leader, and my waypoints just disapeared from my map... That will be fixed in the next upadte ? Have you a workaround to store the AI's waypoints and reload it after fastroping ?

I also noticed a glitch which apears somtimes during AI fatroping: one of the AIs stay grabbed to the cord even if is tuching ground... This bug block the rest of the mission beacause the helicopter does'nt complete his waypoint and stay hovering forever with the unit grabbed to the cord...

Thanks for help and have a nice day.

**Comment by kripto202** [comment]
I'm still trying to figure out how to get fast roping to work on custom models but have had no luck. Can someone tell me how to make sure to get it working properly, including the module itself?

**Comment by bux578** [comment]
@NoBullshit 
That's a todo list issue ... please open a separate issue

@kripto202 might be better if you join our public chat: http://slackin.ace3mod.com/ and ask your questions there

**Comment by NoBullshit** [comment]
**@bux578**
_Please open a separate issue_

Done! (#3687)

**Comment by madpat3** [comment]
First you have to add this to the config.bin of your model:

ace_fastroping_enabled = 2;

ace_fastroping_friesType = "yourFRIESType";

ace_fastroping_friesAttachmentPoint[] = {x, y, z};

ace_fastroping_onCut = "ace_fastroping_fnc_onCutCommon";

ace_fastroping_onPrepare "ace_fastroping_fnc_onPrepareCommon";

ace_fastroping_ropeOrigins[] = {"ropeOriginLeft", "ropeOriginRight"};

then you have to sync the FRIES –Module with that vehicle in the EDITOR and you schould be good to go.

 

 

Von: kripto202 [mailto:notifications@github.com] 
Gesendet: Montag, 11. April 2016 22:09
An: acemod/ACE3
Cc: madpat3
Betreff: Re: [acemod/ACE3] TODO list for Fast Roping (#3475)

 

I'm still trying to figure out how to get fast roping to work on custom models but have had no luck. Can someone tell me how to make sure to get it working properly, including the module itself?

**Comment by Twelvecaliber** [comment]
I apologize if this is not the appropriate place to do this. But has anyone whom has been working on this, seen this mod made by duda: https://forums.bistudio.com/topic/192126-advanced-rappelling/ and could it be considered getting in contact with him, as i think it's fair to say that he has produced some high quality rope related mods. Just a suggestion as he seems rather knowledgeable.

**Comment by BaerMitUmlaut** [comment]
Feature requests should go into #3594. Something like rapelling was already requested there.

**Comment by graffo** [comment]
there is the possibility of implementing a way for the player to select which side of the helicopter he will descend by the rope?

**Comment by GoldJohnKing** [comment]
Feature request:
Would you add a param of auto add fastrope system on Zeus spawned helis? Manually spawn the fastrope system on a heli is truly inconvience in most situations.

**Comment by Drofseh** [comment]
Feature requests should go into #3594, not be comments on old issues.

**Comment by jonpas** [comment]
> Maybe add option to manually retract FRIES

I added that, didn't I?

**Comment by Mike-MF** [comment]
Closing due to age, inactivity or already solved. May be re-opened if the issue is still present, even better just open a new up-to-date Issue.
